test/cmd/apply.sh

```diff
@@ -46,18 +46,18 @@ run_kubectl_apply_tests() {
   # Post-Condition: deployment "test-deployment-retainkeys" created
   kube::test::get_object_assert deployments "{{range.items}}{{$id_field}}{{end}}" 'test-deployment-retainkeys'
   # Post-Condition: deployment "test-deployment-retainkeys" has defaulted fields
-  [[ "$(kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]}" | grep RollingUpdate)" ]]
-  [[ "$(kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]}" | grep maxSurge)" ]]
-  [[ "$(kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]}" | grep maxUnavailable)" ]]
-  [[ "$(kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]}" | grep emptyDir)" ]]
+  kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]}" | grep -q RollingUpdate
+  kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]}" | grep -q maxSurge
+  kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]}" | grep -q maxUnavailable
+  kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]}" | grep -q emptyDir
```

Comment:
BTW is it intentionally that we execute the same command 4 times? It seems that we can execute it only once and later use its output for our checks. Something like this:
```bash
local deployment_yaml="$(kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]}" || true)"
echo "$deployment_yaml" | grep -q RollingUpdate
echo "$deployment_yaml" | grep -q maxSurge
echo "$deployment_yaml" | grep -q maxUnavailable
echo "$deployment_yaml" | grep -q emptyDir
```

Reply:
Good point, will update in other pr since this will be closed as a dup of https://github.com/kubernetes/kubernetes/pull/73518